### PR TITLE
RavenDB-6919: [Linux] MemoryInfo and ProcessorCount take into account cgroup limits

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Raven.Server/Program.cs [hunk 1]
```diff
@@ -218,6 +219,13 @@ private static bool RunInteractive(RavenServer server)
 
                         break;
 
+                    case "info":
+                        var meminfo = MemoryInformation.GetMemoryInfo();
+                        Console.WriteLine(" Build {0}, Version {1}, SemVer {2}, Commit {3}\r\n PID {4}, {5} bits, {6} Cores\r\n {7} Physical Memory, {8} Available Memory",
```
Comment (ayende) [inline src/Raven.Server/Program.cs:224]: Need to make this more readable, units in KB, with thousands separator.

File: src/Sparrow/LowMemory/MemoryInformation.cs [hunk 2]
```diff
@@ -53,31 +53,47 @@ public static unsafe MemoryInfoResult GetMemoryInfo()
...
-                            _logger.Info("Failure when trying to read memory info from posix, error code was: " + Marshal.GetLastWin32Error());
+                        if (Logger.IsInfoEnabled)
+                            Logger.Info("Failure when trying to read memory info from posix, error code was: " + Marshal.GetLastWin32Error());
                         return FailedResult;
                     }
 
+                    Size availableRam, totalPhysicalMemory;
+                    if (cgroupMemoryLimit < (long)info.TotalRam)
+                    {
+                        availableRam = new Size(cgroupMemoryLimit - cgroupMemoryUsage, SizeUnit.Bytes);
```
Comment (ayende) [inline src/Sparrow/LowMemory/MemoryInformation.cs:80]: can this be negative ?

File: src/Sparrow/Platform/Posix/CgroupUtils.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                        coresCount++;
+                    else
+                        coresCount += Convert.ToInt32(cores[1]) - Convert.ToInt32(cores[0]) + 1;  // "5-7" == 3
+                }
+                return coresCount;
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info("Unable to read and parse '{filename}', will not resepect container's number of cores", e);
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/CgroupUtils.cs:53]: Needs to be operations

File: src/Sparrow/Platform/Posix/CgroupUtils.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                var txt = File.ReadAllText(filename);
+                var result = Convert.ToInt64(txt);
+                if (result < 0)
+                    result = long.MaxValue;
+                return result;
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info("Unable to read and parse '{filename}', will not resepect container's limit", e);
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/CgroupUtils.cs:25]: Need to be operations (But only once, I think)

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Program.cs (hunk 1 above)
Comment (aviviadi) [reply src/Raven.Server/Program.cs:224]: This is an output of class Size.  i.e. : "15.59GBytes".  Isn't it readable?

File: src/Sparrow/LowMemory/MemoryInformation.cs (hunk 2 above)
Comment (aviviadi) [reply src/Sparrow/LowMemory/MemoryInformation.cs:80]: yes, when not applicable, it returns -1

File: src/Raven.Server/Program.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Program.cs:224]: `AvailableMemory` is `Size`, sorry. Thought it was raw bytes

File: src/Sparrow/Platform/Posix/CgroupUtils.cs (hunk 4 above)
Comment (aviviadi) [reply src/Sparrow/Platform/Posix/CgroupUtils.cs:25]: https://github.com/ravendb/ravendb/pull/2374

**Pull request merged**